pkg/f1/testing/t.go

```diff
@@ -27,36 +31,70 @@ type T struct {
 func NewT(iter, scenarioName string) (*T, func()) {
 	t := &T{
 		Iteration:     iter,
-		Log:           log.WithField("i", iter).WithField("scenario", scenarioName).Logger,
+		Logger:        log.WithField("i", iter).WithField("scenario", scenarioName).Logger,
 		Scenario:      scenarioName,
 		teardownStack: []func(){},
 	}
 	t.Require = require.New(t)
 	return t, t.teardown
 }
 
-func (t *T) Errorf(format string, args ...interface{}) {
-	atomic.StoreInt64(&t.failed, int64(1))
-	t.Log.Errorf(format, args...)
+// Name returns the name of the running Scenario.
+func (t *T) Name() string {
+	return t.Scenario
 }
 
+// FailNow marks the function as having failed and stops its execution by calling runtime.Goexit (which then runs all deferred calls in the current goroutine).
+// Execution will continue at the next Scenario iteration. FailNow must be called from the goroutine running the Scenario,
+// not from other goroutines created during the Scenario. Calling FailNow does not stop those other goroutines.
 func (t *T) FailNow() {
 	atomic.StoreInt64(&t.failed, int64(1))
-	t.Log.Errorf("%s failed and stopped", t.Iteration)
```

Comment:
Why take out the logging? How will we know that the iteration has failed?

Reply:
We just tried to make it consistent with https://golang.org/pkg/testing/#T.FailNow and because we exposed some logging methods, the client can chose to log separately.